File: ODOT.ARMS.Web/DTOs/ProjectInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace ODOT.ARMS.Web.DTOs
{
    public class ProjectInfo
    {
        public int? ProjectClassificationId { get; set; }
        public int? ProjectStatusId { get; set; }
        public string ProjectTitleTxt { get; set; }
        public string RfpNum { get; set; }
        public string IdeaNum { get; set; }
        public string FedAuthNum { get; set; }
        public string StateJobNum { get; set; }
        public string AgreementNum { get; set; }
        public string PidNum { get; set; }
        public string PropFiscalYr { get; set; }
        public string TpfNum { get; set; }
        public DateTime? ContractStartDt { get; set; }
        public DateTime? ContractEndDt { get; set; }
        public int? AgencyId { get; set; }
        public string VendorIdTxt { get; set; }
        public string AddressCd { get; set; }
        public string ImpStatusInd { get; set; }
        public string StandardDeliverableInd { get; set; }
        public int? ProjectDuration { get; set; }
        public DateTime? CurrentEndDt { get; set; }
        public string GoalsTxt { get; set; }
        public string ActiveInd { get; set; }
        public string ProjId { get; set; }
        public string UserId { get; set; }
        public Int64? WithholdingAmount { get; set; }
        public DateTime EntryDt { get; set; }
        public string ObjectiveTxt { get; set; }
        [DatabaseGeneratedAttribute(DatabaseGeneratedOption.Identity)]
        public int? ProjectAltId { get; set; }
        public ICollection<ProjectType> ProjectTypeList { get; set; }
    }
}

File: ODOT.ARMS.Web/ClientApp/src/app/project/models/projectInfo.ts


export interface IProjectInfo {
  projectAltId: number;
  projectClassificationId: number;
  projectStatusId: number;
  projectStatusTxt: string;
  projectTitleTxt: string;
  rfpNum: string;
  ideaNum: string;
  fedAuthNum: string;
  stateJobNum: string;
  agreementNum: string;
  pidNum: string;
  propFiscalYr: number;
  tpfNum: string;
  contractStartDt: Date;
  contractEndDt: Date;
  agencyId: number;
  agencyName: string;
  vendorIdTxt: string;
  addressCd: string;
  impStatusInd: string;
  standardDeliverableInd: string;
  projectDuration: number;
  currentEndDt: Date;
  goalsTxt: string;
  activeInd: string;
  projId: string;
  userId: string;
  entryDt: Date;
  objectiveTxt: string;
  withholdingAmount: number;
  projectTypeList: ProjectTypeList[];
}
export class ProjectInfo implements IProjectInfo {
  projectAltId: number;
  projectClassificationId: number;
  projectStatusId: number;
  projectStatusTxt: string;
  projectTitleTxt: string;
  rfpNum: string;
  ideaNum: string;
  fedAuthNum: string;
  stateJobNum: string;
  agreementNum: string;
  pidNum: string;
  propFiscalYr: number;
  tpfNum: string;
  contractStartDt: Date;
  contractEndDt: Date;
  agencyId: number;
  agencyName: string;
  vendorIdTxt: string;
  addressCd: string;
  impStatusInd: string;
  standardDeliverableInd: string;
  projectDuration: number;
  currentEndDt: Date;
  goalsTxt: string;
  activeInd: string;
  projId: string;
  userId: string;
  entryDt: Date;
  objectiveTxt: string;
  withholdingAmount: number;
  projectTypeList: ProjectTypeList[];
}

export class ProjectTypeList {
  projectTypeId: string;

  constructor(typeId: string) {
    this.projectTypeId = typeId;
  }

}

export class Addresscode {
  OaksVendorNo: string;
  addressseqno: string;
}

export enum ProjectClassificationEnum {
  STANDARD = 72,
  ROC = 73,
  ORIL = 74,
  POOLED = 75
}

export enum ProjectStatusEnum {
  Proposed = 108,
  Active = 109,
  Completed = 110,
  Cancelled = 111,
  Terminated = 112,
  Hold = 113,
  AwaitingFinalInvoice = 114
}


